app/serializers/api/v1/message_serializer.rb

```diff
@@ -81,5 +81,28 @@ def state__sql
          END"
       end
     end
+
+    def unread_count
```

Comment:
I'm not sure if this is the right place to put `unread` count... This serializer is for individual messages and groups of messages but not for aggregate data like count of unread messages. This would also explain why `unread_count__sql` wasn't right.

One way we have handled this in the past is to attach use `meta` payloads that go alongside the serializers.

```
{
  meta: { 
      unread_messages_count: 3
  },
  data: {
    ...
  }
}
```

Reply:
Added separate serializer for message notifications.